examples/pytorch/ogc/utils.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+import scipy.sparse as sp
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+
+
+def sparse_mx_to_torch_sparse_tensor(sparse_mx):
+    """Convert a scipy sparse matrix to a torch sparse tensor."""
+    sparse_mx = sparse_mx.tocoo().astype(np.float32)
+    indices = torch.from_numpy(
+        np.vstack((sparse_mx.row, sparse_mx.col)).astype(np.int64)
+    )
+    values = torch.from_numpy(sparse_mx.data)
+    shape = torch.Size(sparse_mx.shape)
+    return torch.sparse.FloatTensor(indices, values, shape)
+
+
+def symmetric_normalize_adjacency(graph):
+    """Symmetric normalize graph adjacency matrix."""
+    adj = graph.adjacency_matrix()
+    in_degs = graph.in_degrees().float()
+    in_norm = torch.pow(in_degs, -0.5).unsqueeze(-1)
+    degi = torch.diag(torch.squeeze(torch.t(in_norm)))
```

Comment:
You could probably just do `in_norm.T` instead of `torch.t`.

Reply:
Thanks. As suggested, we have changed it.
And we've completely overhauled our codebase, adopting a 'gnn-pipeline' style. The updated version comprises three files: 'train.py' serves as the main file, 'ogc.py' houses the employed model, and 'utils.py' encompasses all fundamental method components.
Furthermore, in this part, we have refined the normalization process, aligning it with the example code in 'dgl/../pytorch/gcn'.